### DISCUSSION TITLE
Streaming mcap remotely

### DISCUSSION THREAD

**Discussion by PavanChillakuru** [foxglove/community#33] [Q&A]
**Description**
Currently we are exploring fox glove studio, Is there any way to stream remote url file directly, rather loading file completely and playing?

- Version:
- OS:
- Data source ( mcap file):
- ROS distro (if applicable):

**Steps To Reproduce**

**Current Experience**
Studio is downloading the file from remote_url and then start playing.

**Desired Experience**
It should stream mcap file remotely.

**Comment by amacneil** [comment]
When you open a remote URL, it should only download parts of the mcap file as needed. What makes you think it is loading the file completely?

**Reply by PavanChillakuru** [reply]
Thanks for your prompt reply, studio actually playing the file after loading for a while, also when we checked in network console, its showing that its loading file first, please refer attached screenshot below
![Screenshot from 2022-07-26 19-57-05](https://user-images.githubusercontent.com/53422765/181034824-0a8220c8-7272-4636-b510-238099bc990a.png)
.

**Reply by amacneil** [reply]
I see. Can you try enabling "Experimental mcap player" in the settings?

We will make this new player the default very soon.

**Reply by PavanChillakuru** [reply]
After enabling "Experimental mcap player" its same behaviour.
<img width="1191" alt="Screenshot 2022-07-27 at 12 29 52 AM" src="https://user-images.githubusercontent.com/53422765/181090101-be032d6c-c3b3-45f3-952e-e7c8b8190258.png">

**Reply by jtbandes** [reply]
Can you show the output of `mcap info` on this file? I would be interested to learn about how the chunks are set up in the file. If you click on the 3rd (longest) request in the debugger, can you show the request headers? This should reveal whether the app is requesting the whole file or a short range.

**Comment by defunctzombie** [comment]
The experimental mcap player does not yet work for remote files. I opened an issue to track adding this: https://github.com/foxglove/studio/issues/3949
